ocaml/xapi/quicktest_bv_tests.ml

```diff
@@ -0,0 +1,94 @@
+open Quicktest_common
+
+let start session_id rpc =
+
+  (* helper functions for API/PBD calls *)
+  let with_api fn = fn ~rpc ~session_id in
+  let with_obj fn obj = with_api fn ~self:obj in
+
+  (* with_obj generalises to most classes, however, for some classes
+   * (SR,Host,VDI,etc), a mix of label ~self and ~sr,~vdi etc is used *)
+
+let get_set_config (class_name, obj, gt, st, permission) =
+    let conf_t = make_test ("Testing " ^ class_name ^ " 'get,set_other_config'") 4 in
+    try
+      (* set config to arbitrary (key,value) *)
+      start conf_t;
+      let dummy_value = [("dummy_key","dummy_val")] in
+      with_obj st obj ~value:dummy_value; (* set config value *)
+
+      (* read-writeable configs succeed if the value is set, read-only fails *)
+      if dummy_value = (with_obj gt obj)
+      then match permission with
+        | `RW -> success conf_t
+        | `RO -> failed conf_t "Error: config read-only, should throw PERMISSION_DENIED"
+      else failed conf_t "set_config failed";
+
+      (* read-only config returns success test if and only if it throws PERMISSION_DENIED *)
+    with
+      | (Failure hd) -> failed conf_t (Printf.sprintf "Error: could not find %s object" class_name)
+      | Api_errors.Server_error (x,_) when x = Api_errors.permission_denied && permission =`RO -> success conf_t;
+      | e -> failed conf_t (ExnHelper.string_of_exn e); in
+
+
+  (* check add_to and remove_from_other_config *)
+  let add_rm_other_config (class_name, obj, gt, add, rm)  =
+    let adrm_t = make_test ("Testing " ^ class_name ^ " 'add_to/remove_from_other_config'") 4 in
+    try
+      start adrm_t;
+
+      (* add ("key","value") to other_config *)
+      let key,value = "key","value" in
+      with_obj add obj ~key ~value;
+
+      (* remove from other config *)
+      if List.mem (key,value) (with_obj gt obj)
+      then with_obj rm obj ~key (* may need ignore *)
+      else failed adrm_t "Error: add_to_other_config failed";
+
+      (* test fails if (key,value) has not been removed *)
+      if List.mem (key,value) (with_obj gt obj)
```

Comment:
Please add 2 keys and remove one, checking that the other is still present as well